### DISCUSSION TITLE
Gradient field on vertices instead of faces

### DISCUSSION THREAD

**Discussion by avaruuser** [libigl/libigl#1948] [Q&A] [answered]
Hi all,

Does libigl implement a neat way to compute the gradient operator on the vertices of a mesh instead of on the faces? Or are there implementations of methods to interpolate face-based vectors to vertex-based vectors? I couldn't find any such functions in libigl and it appears to be more involved in general (at least for someone who's not well-versed in differential geometry).

I would appreciate any help with this!

**Comment by alecjacobson** [comment, marked as answer]
You could compute gradients onto faces and then average onto vertices with https://github.com/libigl/libigl/blob/main/include/igl/average_onto_vertices.h

If you're starting with per-vertex quantities and you need per-vertex gradients, then this may require interpreting your per-vertex quantities in a non-standard way (if standard is a piecewise-linear function then it's gradient is piecewise constant at faces and not defined at vertices). Personally, I'd recommend 1) re-investigate whether you really need gradients at vertices and if so 2) try simple averaging first.
